### ISSUE TITLE
Slack requires an api (instead of bot) token for joining channels

### ISSUE THREAD

**Issue by jlsjonas** [42wim/matterbridge#75]
*stable release* process just idling (still running) after a while, not forwarding any message anymore
(dev branch wouldn't forward slack messages, only mattermost messages)

**Comment by 42wim** [comment]
@jlsjonas Please test/report with master, i'm planning on release v0.7.0 today or tomorrow.
0.7 has fixed a lot of issues regarding stability

**Comment by jlsjonas** [comment]
> (dev branch wouldn't forward slack messages, only mattermost messages)

although that issue seemed to be limited to os x somehow, testing with 0.7 (on linux) at the moment

**Comment by 42wim** [comment]
You need to create a seperate bot user, just like in mattermost. This is to prevent loops.

**Comment by jlsjonas** [comment]
yes, that's how it currently set up (and working)
however the "proper" way would be using an api token to join channels (since bot tokens don't have permissions to join channels), and using a "bot user"

**Comment by 42wim** [comment]
I'm going to revert the change so that you don't need to setup a separate bot account, you can use your own test-token.
(I think the looping issue was while I was testing, shouldn't happen in normal use)

I'm not really at home with slack tokens, I know of the following:
* bot token - limited rights, cannot join channels
* test token - token for your own (real) account

Is the api token you are talking about the "test token" or something else ?

**Comment by jlsjonas** [comment]
the current (working) setup is using a test-token, however these are basically just an api token with the (deprecated) "client" permission, the current way of "properly" doing it would be to have an api token (basically the same as a test token, but more appropriate scope) for the joinChannel command, and a bot token for chat

the bot tokens are a little ambiguous in the way that there's a "custom bot" (which isn't linked to an api token), made for internal usage; and a "more limited" (though everything that is limited can&should be done by regular api instead, and except for history is outside the (future) scope of this project)

TL;DR: ideally accept an api token (for joinChannel) and a bot token (for chat bridging/mirroring)

**Comment by 42wim** [comment]
Thanks for the explanation. 
I'll update the documentation for now to make it clear to use a test-token.

**Comment by lululombard** [comment]
The creation of legacy test tokens is deprecated as of May 2020, is there an alternative? Thanks https://api.slack.com/changelog/2020-02-legacy-test-token-creation-to-retire

**Comment by 42wim** [comment]
@lululombard see https://github.com/42wim/matterbridge/wiki/Slack-bot-setup
